### ISSUE TITLE
The RF stops at the breakpoint, but when I step into the keyword RED does not show the file of the keyword that is stepped in

### ISSUE THREAD

**Issue by KrzysztofJozefowicz** [nokia/RED#74]
@danielvanmil comments from #59 copied to new issue:
Daniel:
I have a project here with a breakpoint. The RF stops at the breakpoint, but when I step into the keyword RED does not show the file of the keyword that is stepped in, although the debug view shows the correct statements that are executed when stepping further into the keyword. So it seems all goes well, but only the view of the current file is not correctly updated. I cannot find the cause of the issue by creating a simple project with the same structure etc.

Any hints on how to debug the resolving of the files etc. or is debugging with the RED source code the only option. Any logging about the resolving?

Thanks in advance.

KJ:
Hi
to be clear, the keyword which you would like to enter is an Robot defined Keyword,not an keyword imported from Python library?
Can you provide more specific details (keyword name, where it is used in testcase,project structure etc.) so we could investigate it further?

Daniel:
yes, the keyword is a Robot defined keyword. I'm trying to define the exact use case to reproduce the issue, but I cannot reproduce it yet with a clean example project with the same structure as mentioned. The project it appears is too big and complicated to hand over to you.

Maybe attaching a debugger (to Eclipse) and quickly debug the code of this issue to find the cause and reproduce the issue? Does this look like the best plan for now maybe?

KJ:
I will get to your about how to reproduce the case on small project, also we have observed that there are problems when keyword is called in Suite Setup ,Teardown and --init-- files. Does your project files are located inside workspace on disk or linked inside Project/Workspace?

**Comment by danielvanmil** [comment]
the project is located outside the workspace. The small test project is located inside the workspace. Can this affect the behavior? I can test the small test project outside of the workspace.

**Comment by KrzysztofJozefowicz** [comment]
Hi
we have a test to verify that Debug works with projects outside workspace,any more specific information would be needed, 
Can you verify below :
1. Keyword is used in Suite setup
2. Keyword is used in Teardown
3. Keyword is used in -- init -- 
4. Does a name of KW consists periods (not uses as resource/library prefix)

**Comment by danielvanmil** [comment]
- moved project to workspace: still the same issue. I cannot seem to reproduce the use case that causes the issue.
- cloned the RED repo and attached a debugger that's working

Any hints how to check (where to set breakpoints etc.) what's not functioning and what logic should be invoked to switch to the correct keyword (file).

Thanks.

**Comment by modrz** [comment]
Do you use any __init__.robot file in your project?
If so debugger may not work correctly, because of a bug existing since version 0.7.3
It will be fixed in version 0.7.5

**Comment by danielvanmil** [comment]
yes, we have more init files in our project. Upgraded to 0.7.5, but now debugging is extremely slow at some point and doesn't seem to stop at a breakpoint anymore. Stack dump during the moments the test seems to hang:

                                                                                                                                                                                                                                                                                                                            2017-02-08 12:46:57
Full thread dump Java HotSpot(TM) 64-Bit Server VM (25.40-b25 mixed mode):

"Worker-29" #141 prio=5 os_prio=0 tid=0x000000001a831800 nid=0x26ac in Object.wait() [0x00000000365df000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:188)
        - locked <0x000000008001e500> (a org.eclipse.core.internal.jobs.WorkerPool)
        at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:220)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:52)

"Timer-13" #140 daemon prio=6 os_prio=0 tid=0x000000001a828800 nid=0x6e14 in Object.wait() [0x00000000364df000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at java.util.TimerThread.mainLoop(Unknown Source)
        - locked <0x00000000862ca490> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Unknown Source)

"Timer-12" #139 daemon prio=6 os_prio=0 tid=0x000000001a82f000 nid=0x4570 in Object.wait() [0x00000000363df000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at java.util.TimerThread.mainLoop(Unknown Source)
        - locked <0x00000000862ca930> (a java.util.TaskQueue)
        at java.util.TimerThread.run(Unknown Source)

"Process monitor" #138 daemon prio=5 os_prio=0 tid=0x000000001a82a000 nid=0x4160 runnable [0x00000000362df000]
   java.lang.Thread.State: RUNNABLE
        at java.lang.ProcessImpl.waitForInterruptibly(Native Method)
        at java.lang.ProcessImpl.waitFor(Unknown Source)
        at org.eclipse.debug.core.model.RuntimeProcess$ProcessMonitorThread.run(RuntimeProcess.java:428)

"Input Stream Monitor" #137 daemon prio=5 os_prio=0 tid=0x000000001f6c2800 nid=0x76b0 in Object.wait() [0x00000000361df000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.debug.internal.core.InputStreamMonitor.writeNext(InputStreamMonitor.java:156)
        - locked <0x00000000862cad98> (a java.lang.Object)
        at org.eclipse.debug.internal.core.InputStreamMonitor.write(InputStreamMonitor.java:125)
        at org.eclipse.debug.internal.core.InputStreamMonitor$1.run(InputStreamMonitor.java:100)
        at java.lang.Thread.run(Unknown Source)

"Output Stream Monitor" #136 daemon prio=1 os_prio=-2 tid=0x000000001f6be000 nid=0x5388 runnable [0x00000000360df000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(Unknown Source)
        at java.io.BufferedInputStream.read1(Unknown Source)
        at java.io.BufferedInputStream.read(Unknown Source)
        - locked <0x00000000862cb068> (a java.io.BufferedInputStream)
        at java.io.FilterInputStream.read(Unknown Source)
        at org.eclipse.debug.internal.core.OutputStreamMonitor.read(OutputStreamMonitor.java:147)
        at org.eclipse.debug.internal.core.OutputStreamMonitor.access$1(OutputStreamMonitor.java:137)
        at org.eclipse.debug.internal.core.OutputStreamMonitor$1.run(OutputStreamMonitor.java:212)
        at java.lang.Thread.run(Unknown Source)

"Output Stream Monitor" #135 daemon prio=1 os_prio=-2 tid=0x000000001f6bb800 nid=0x70f4 runnable [0x0000000035fdf000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(Unknown Source)
        at java.io.BufferedInputStream.read1(Unknown Source)
        at java.io.BufferedInputStream.read(Unknown Source)
        - locked <0x00000000862c8378> (a java.io.BufferedInputStream)
        at java.io.BufferedInputStream.read1(Unknown Source)
        at java.io.BufferedInputStream.read(Unknown Source)
        - locked <0x00000000862c8738> (a java.io.BufferedInputStream)
        at java.io.FilterInputStream.read(Unknown Source)
        at org.eclipse.debug.internal.core.OutputStreamMonitor.read(OutputStreamMonitor.java:147)
        at org.eclipse.debug.internal.core.OutputStreamMonitor.access$1(OutputStreamMonitor.java:137)
        at org.eclipse.debug.internal.core.OutputStreamMonitor$1.run(OutputStreamMonitor.java:212)
        at java.lang.Thread.run(Unknown Source)

"Worker-28" #133 prio=5 os_prio=0 tid=0x000000001f6bd000 nid=0x693c in Object.wait() [0x0000000035ddf000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:188)
        - locked <0x000000008001e500> (a org.eclipse.core.internal.jobs.WorkerPool)
        at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:233)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:52)

"Worker-27" #132 prio=5 os_prio=0 tid=0x000000001f6bc800 nid=0x50ec runnable [0x0000000035cdf000]
   java.lang.Thread.State: RUNNABLE
        at java.io.WinNTFileSystem.getBooleanAttributes(Native Method)
        at java.io.File.isFile(Unknown Source)
        at org.rf.ide.core.testdata.model.RobotFileOutput.getType(RobotFileOutput.java:306)
        at org.rf.ide.core.execution.context.UserKeywordExecutableRowFinder$UserKeywordExtractor.scope(UserKeywordExecutableRowFinder.java:199)
        at org.rf.ide.core.execution.context.UserKeywordExecutableRoawFinder$UserKeywordExtractor.scope(UserKeywordExecutableRowFinder.java:195)
        at org.rf.ide.core.testdata.model.search.keyword.KeywordSearcher.findKeywords(KeywordSearcher.java:104)
        at org.rf.ide.core.execution.context.UserKeywordExecutableRowFinder.findExecutableRow(UserKeywordExecutableRowFinder.java:59)
        at org.rf.ide.core.execution.context.KeywordPositionFinder.findExecutableRow(KeywordPositionFinder.java:45)
        at org.rf.ide.core.execution.context.KeywordPositionFinder.find(KeywordPositionFinder.java:36)
        at org.rf.ide.core.execution.context.RobotDebugExecutionContext.findKeywordPosition(RobotDebugExecutionContext.java:109)
        at org.robotframework.ide.eclipse.main.plugin.debug.RobotDebugEventDispatcher.handleStartKeywordEvent(RobotDebugEventDispatcher.java:231)
        at org.robotframework.ide.eclipse.main.plugin.debug.RobotDebugEventDispatcher.runEventsLoop(RobotDebugEventDispatcher.java:122)
        at org.robotframework.ide.eclipse.main.plugin.debug.RobotDebugEventDispatcher.run(RobotDebugEventDispatcher.java:87)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

"Worker-26" #131 prio=5 os_prio=0 tid=0x000000001f6c0000 nid=0x28bc in Object.wait() [0x0000000035bdf000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:188)
        - locked <0x000000008001e500> (a org.eclipse.core.internal.jobs.WorkerPool)
        at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:220)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:52)

"Worker-25" #130 prio=5 os_prio=0 tid=0x000000001f6c1800 nid=0x77c0 runnable [0x0000000035adf000]
   java.lang.Thread.State: RUNNABLE
        at java.lang.ProcessImpl.waitForInterruptibly(Native Method)
        at java.lang.ProcessImpl.waitFor(Unknown Source)
        at org.robotframework.ide.eclipse.main.plugin.launch.RobotLaunchConfigurationDelegate.doLaunch(RobotLaunchConfigurationDelegate.java:256)
        at org.robotframework.ide.eclipse.main.plugin.launch.RobotLaunchConfigurationDelegate.launch(RobotLaunchConfigurationDelegate.java:156)
        at org.eclipse.debug.internal.core.LaunchConfiguration.launch(LaunchConfiguration.java:885)
        at org.eclipse.debug.internal.core.LaunchConfiguration.launch(LaunchConfiguration.java:739)
        at org.eclipse.debug.internal.ui.DebugUIPlugin.buildAndLaunch(DebugUIPlugin.java:1039)
        at org.eclipse.debug.internal.ui.DebugUIPlugin$8.run(DebugUIPlugin.java:1256)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

"Worker-24" #127 prio=5 os_prio=0 tid=0x000000001f6c1000 nid=0x676c in Object.wait() [0x00000000357df000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.WorkerPool.sleep(WorkerPool.java:188)
        - locked <0x000000008001e500> (a org.eclipse.core.internal.jobs.WorkerPool)
        at org.eclipse.core.internal.jobs.WorkerPool.startJob(WorkerPool.java:220)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:52)

"Thread-24" #90 prio=5 os_prio=0 tid=0x000000001a827000 nid=0x5cf8 waiting on condition [0x00000000330df000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x0000000085550b98> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(Unknown Source)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(Unknown Source)
        at java.util.concurrent.LinkedBlockingQueue.take(Unknown Source)
        at org.rf.ide.core.fileWatcher.RedFileWatcher$2.run(RedFileWatcher.java:179)
        at java.lang.Thread.run(Unknown Source)

"Thread-23" #89 prio=5 os_prio=0 tid=0x000000001a82d800 nid=0x621c waiting on condition [0x0000000032fdf000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000000855500e0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.park(Unknown Source)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(Unknown Source)
        at java.util.concurrent.LinkedBlockingDeque.takeFirst(Unknown Source)
        at java.util.concurrent.LinkedBlockingDeque.take(Unknown Source)
        at sun.nio.fs.AbstractWatchService.take(Unknown Source)
        at org.rf.ide.core.fileWatcher.RedFileWatcher$1.run(RedFileWatcher.java:134)
        at java.lang.Thread.run(Unknown Source)

"Thread-22" #88 daemon prio=5 os_prio=0 tid=0x000000001a82d000 nid=0x5530 runnable [0x0000000032edf000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus0(Native Method)
        at sun.nio.fs.WindowsNativeDispatcher.GetQueuedCompletionStatus(Unknown Source)
        at sun.nio.fs.WindowsWatchService$Poller.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

"Thread-21" #86 prio=5 os_prio=0 tid=0x000000001ffc7000 nid=0x7a60 runnable [0x0000000032bdf000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(Unknown Source)
        at sun.nio.cs.StreamDecoder.readBytes(Unknown Source)
        at sun.nio.cs.StreamDecoder.implRead(Unknown Source)
        at sun.nio.cs.StreamDecoder.read(Unknown Source)
        - locked <0x00000000855bf938> (a java.io.InputStreamReader)
        at java.io.InputStreamReader.read(Unknown Source)
        at java.io.BufferedReader.fill(Unknown Source)
        at java.io.BufferedReader.readLine(Unknown Source)
        - locked <0x00000000855bf938> (a java.io.InputStreamReader)
        at java.io.BufferedReader.readLine(Unknown Source)
        at org.rf.ide.core.executor.RobotCommandRcpExecutor$2.run(RobotCommandRcpExecutor.java:150)
        at java.lang.Thread.run(Unknown Source)

"Thread-20" #85 prio=5 os_prio=0 tid=0x000000001ffcc000 nid=0x2774 runnable [0x0000000032ade000]
   java.lang.Thread.State: RUNNABLE
        at java.io.FileInputStream.readBytes(Native Method)
        at java.io.FileInputStream.read(Unknown Source)
        at java.io.BufferedInputStream.read1(Unknown Source)
        at java.io.BufferedInputStream.read(Unknown Source)
        - locked <0x00000000855253c8> (a java.io.BufferedInputStream)
        at sun.nio.cs.StreamDecoder.readBytes(Unknown Source)
        at sun.nio.cs.StreamDecoder.implRead(Unknown Source)
        at sun.nio.cs.StreamDecoder.read(Unknown Source)
        - locked <0x00000000855a4c90> (a java.io.InputStreamReader)
        at java.io.InputStreamReader.read(Unknown Source)
        at java.io.BufferedReader.fill(Unknown Source)
        at java.io.BufferedReader.readLine(Unknown Source)
        - locked <0x00000000855a4c90> (a java.io.InputStreamReader)
        at java.io.BufferedReader.readLine(Unknown Source)
        at org.rf.ide.core.executor.RobotCommandRcpExecutor$1.run(RobotCommandRcpExecutor.java:120)
        at java.lang.Thread.run(Unknown Source)

"Worker-16" #68 prio=5 os_prio=0 tid=0x000000001a826000 nid=0x5f94 in Object.wait() [0x00000000317df000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.ui.console.IOConsoleInputStream.waitForData(IOConsoleInputStream.java:150)
        at org.eclipse.ui.console.IOConsoleInputStream.read(IOConsoleInputStream.java:93)
        - locked <0x00000000862dd040> (a org.eclipse.ui.console.IOConsoleInputStream)
        at org.eclipse.ui.console.IOConsoleInputStream.read(IOConsoleInputStream.java:119)
        at org.eclipse.debug.internal.ui.views.console.ProcessConsole$InputReadJob.run(ProcessConsole.java:733)
        at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)

"[ThreadPool Manager] - Idle Thread" #40 daemon prio=5 os_prio=0 tid=0x000000001a82b800 nid=0xfcc in Object.wait() [0x000000002e8ff000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.equinox.internal.util.impl.tpt.threadpool.Executor.run(Executor.java:106)
        - locked <0x0000000083614130> (a org.eclipse.equinox.internal.util.impl.tpt.threadpool.Executor)

"Provisioning Event Dispatcher" #39 daemon prio=5 os_prio=0 tid=0x000000001a827800 nid=0x6fa0 in Object.wait() [0x000000002df8f000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x00000000831767a0> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"Bundle File Closer" #34 daemon prio=6 os_prio=0 tid=0x000000001ffc9000 nid=0x5aec in Object.wait() [0x000000002a3cf000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x0000000081b2a680> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"org.eclipse.jface.text.reconciler.MonoReconciler" #33 daemon prio=1 os_prio=-2 tid=0x000000001ffc6000 nid=0x6ee0 in Object.wait() [0x000000002a2cf000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.jface.text.reconciler.AbstractReconciler$BackgroundThread.run(AbstractReconciler.java:180)
        - locked <0x00000000815f7720> (a org.eclipse.jface.text.reconciler.DirtyRegionQueue)

"EventConflaterChain-1" #32 prio=6 os_prio=0 tid=0x000000001ffca800 nid=0x3f04 runnable [0x0000000029b4f000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x0000000081830d80> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.getTask(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

"GlazedListsEventLayer-1" #31 prio=6 os_prio=0 tid=0x000000001ffc8800 nid=0x43f8 runnable [0x0000000029a4f000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000000818312b0> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.getTask(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

"EventAdmin Async Event Dispatcher Thread" #27 daemon prio=6 os_prio=0 tid=0x000000001ffcb800 nid=0x7a54 in Object.wait() [0x000000002533f000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x0000000081b778f8> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"EMF Reference Cleaner" #23 daemon prio=6 os_prio=0 tid=0x000000002023e800 nid=0x251c in Object.wait() [0x00000000213ef000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.ref.ReferenceQueue.remove(Unknown Source)
        - locked <0x00000000808c9ca0> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(Unknown Source)
        at org.eclipse.emf.common.util.CommonUtil$1ReferenceClearingQueuePollingThread.run(CommonUtil.java:70)

"Worker-JM" #18 prio=5 os_prio=0 tid=0x000000001a95a800 nid=0x3cf4 in Object.wait() [0x000000001cbef000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.core.internal.jobs.InternalWorker.run(InternalWorker.java:59)
        - locked <0x000000008001fd20> (a java.util.ArrayList)

"[Timer] - Main Queue Handler" #17 daemon prio=5 os_prio=0 tid=0x000000001a5a6800 nid=0x7820 in Object.wait() [0x000000001c4ef000]
   java.lang.Thread.State: TIMED_WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at org.eclipse.equinox.internal.util.impl.tpt.timer.TimerImpl.run(TimerImpl.java:141)
        - locked <0x00000000800163a0> (a java.lang.Object)
        at java.lang.Thread.run(Unknown Source)

"Start Level: Equinox Container: eb0c4dff-cb1c-471c-8f76-4ee9e2c98910" #15 daemon prio=5 os_prio=0 tid=0x000000001a5f2800 nid=0x5904 in Object.wait() [0x000000001c0ef000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        - waiting on <0x0000000080016c60> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x0000000080016c60> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"Framework Event Dispatcher: org.eclipse.osgi.internal.framework.EquinoxEventPublisher@38467116" #14 daemon prio=5 os_prio=0 tid=0x000000001ae8a800 nid=0x7a80 in Object.wait() [0x000000001bfef000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.getNextEvent(EventManager.java:400)
        - locked <0x000000008002d448> (a org.eclipse.osgi.framework.eventmgr.EventManager$EventThread)
        at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:336)

"Active Thread: Equinox Container: eb0c4dff-cb1c-471c-8f76-4ee9e2c98910" #12 prio=5 os_prio=0 tid=0x000000001aa48800 nid=0x2088 waiting on condition [0x000000001beef000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x000000008002d748> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
        at java.util.concurrent.locks.LockSupport.parkNanos(Unknown Source)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(Unknown Source)
        at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.getTask(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

"Service Thread" #10 daemon prio=9 os_prio=0 tid=0x00000000197c6000 nid=0x3fd0 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C1 CompilerThread3" #9 daemon prio=9 os_prio=2 tid=0x00000000197be000 nid=0x20e4 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread2" #8 daemon prio=9 os_prio=2 tid=0x00000000197bb000 nid=0x68ac waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread1" #7 daemon prio=9 os_prio=2 tid=0x0000000019764800 nid=0x4f48 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"C2 CompilerThread0" #6 daemon prio=9 os_prio=2 tid=0x000000001975d800 nid=0x4374 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Attach Listener" #5 daemon prio=5 os_prio=2 tid=0x0000000019759000 nid=0x4d48 runnable [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Signal Dispatcher" #4 daemon prio=9 os_prio=2 tid=0x0000000019758000 nid=0x51b0 waiting on condition [0x0000000000000000]
   java.lang.Thread.State: RUNNABLE

"Finalizer" #3 daemon prio=8 os_prio=1 tid=0x0000000002669800 nid=0x2bf4 in Object.wait() [0x000000001971f000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.ref.ReferenceQueue.remove(Unknown Source)
        - locked <0x00000000800201a8> (a java.lang.ref.ReferenceQueue$Lock)
        at java.lang.ref.ReferenceQueue.remove(Unknown Source)
        at java.lang.ref.Finalizer$FinalizerThread.run(Unknown Source)

"Reference Handler" #2 daemon prio=10 os_prio=2 tid=0x000000001770d800 nid=0x37e8 in Object.wait() [0x000000001961f000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Unknown Source)
        at java.lang.ref.Reference$ReferenceHandler.run(Unknown Source)
        - locked <0x000000008001def8> (a java.lang.ref.Reference$Lock)

"main" #1 prio=6 os_prio=0 tid=0x000000000200e000 nid=0x377c runnable [0x000000000014e000]
   java.lang.Thread.State: RUNNABLE
        at org.eclipse.swt.internal.win32.OS.WaitMessage(Native Method)
        at org.eclipse.swt.widgets.Display.sleep(Display.java:4788)
        at org.eclipse.ui.application.WorkbenchAdvisor.eventLoopIdle(WorkbenchAdvisor.java:368)
        at org.eclipse.ui.internal.ide.application.IDEWorkbenchAdvisor.eventLoopIdle(IDEWorkbenchAdvisor.java:888)
        at org.eclipse.ui.internal.Workbench$3.eventLoopIdle(Workbench.java:526)
        at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1126)
        at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
        at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
        at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
        at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
        at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
        at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
        at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
        at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
        at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
        at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
        at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
        at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
        at org.eclipse.equinox.launcher.Main.run(Main.java:1519)

"VM Thread" os_prio=2 tid=0x0000000017709800 nid=0x6964 runnable

"GC task thread#0 (ParallelGC)" os_prio=0 tid=0x0000000002587000 nid=0x3738 runnable

"GC task thread#1 (ParallelGC)" os_prio=0 tid=0x0000000002588800 nid=0x2334 runnable

"GC task thread#2 (ParallelGC)" os_prio=0 tid=0x000000000258a000 nid=0x11a4 runnable

"GC task thread#3 (ParallelGC)" os_prio=0 tid=0x000000000258c800 nid=0x1504 runnable

"GC task thread#4 (ParallelGC)" os_prio=0 tid=0x000000000258e800 nid=0x5cf4 runnable

"GC task thread#5 (ParallelGC)" os_prio=0 tid=0x0000000002590000 nid=0x1618 runnable

"GC task thread#6 (ParallelGC)" os_prio=0 tid=0x0000000002593000 nid=0x6db8 runnable

"GC task thread#7 (ParallelGC)" os_prio=0 tid=0x0000000002594000 nid=0x6010 runnable

"VM Periodic Task Thread" os_prio=2 tid=0x00000000197c7000 nid=0x7440 waiting on condition

JNI global references: 772

Heap
 PSYoungGen      total 174080K, used 42261K [0x00000000d5580000, 0x00000000e0000000, 0x0000000100000000)
  eden space 173568K, 24% used [0x00000000d5580000,0x00000000d7e85458,0x00000000dff00000)
  from space 512K, 50% used [0x00000000dff00000,0x00000000dff40000,0x00000000dff80000)
  to   space 512K, 0% used [0x00000000dff80000,0x00000000dff80000,0x00000000e0000000)
 ParOldGen       total 658432K, used 105998K [0x0000000080000000, 0x00000000a8300000, 0x00000000d5580000)
  object space 658432K, 16% used [0x0000000080000000,0x0000000086783ab0,0x00000000a8300000)
 Metaspace       used 81413K, capacity 87572K, committed 87832K, reserved 1124352K
  class space    used 10263K, capacity 12005K, committed 12080K, reserved 1048576K

**Comment by michalanglart** [comment]
Please check if you experience similar issues with 0.8.0
